src/core/algorithms/dd/dd_verifier/dd_verifier.cpp

```diff
@@ -0,0 +1,211 @@
+#include "algorithms/dd/dd_verifier/dd_verifier.h"
+
+#include <chrono>
+#include <cstddef>
+#include <vector>
+
+#include <easylogging++.h>
+
+#include "config/descriptions.h"
+#include "config/names.h"
+#include "config/option_using.h"
+#include "config/tabular_data/input_table/option.h"
+#include "model/table/vertical.h"
+
+namespace algos::dd {
+DDVerifier::DDVerifier() : Algorithm({}) {
+    RegisterOptions();
+    MakeOptionsAvailable({config::kTableOpt.GetName()});
+}
+
+void DDVerifier::RegisterOptions() {
+    DESBORDANTE_OPTION_USING;
+    auto const default_dd = DDs();
+    RegisterOption(config::kTableOpt(&input_table_));
+    RegisterOption(Option{&dd_, kDDString, kDDDString, default_dd});
+}
+
+double DDVerifier::GetError() const {
+    return error_;
+}
+
+void DDVerifier::VisualizeHighlights() {
+    for (auto const &hl : highlights_) {
+        auto const &col_data = typed_relation_->GetColumnData(hl.GetAttributeIndex());
+        auto const col_schema = typed_relation_->GetSchema();
+        auto const &pair = hl.GetPairRows();
+        LOG(DEBUG) << "DD not Holds in " << col_schema->GetColumn(hl.GetAttributeIndex())->GetName()
+                   << " in " << pair.first << " and " << pair.second << " rows with values "
+                   << col_data.GetDataAsString(pair.first) << ", "
+                   << col_data.GetDataAsString(pair.second) << '\n';
+    }
+}
+
+std::size_t DDVerifier::GetNumErrorRhs() const {
+    return num_error_rhs_;
+}
+
+bool DDVerifier::IsColumnMetrizable(model::ColumnIndex const column_index) const {
+    model::TypedColumnData const &column = typed_relation_->GetColumnData(column_index);
+    model::TypeId const type_id = column.GetTypeId();
+
+    if (type_id == +model::TypeId::kUndefined) {
+        throw std::invalid_argument("Column with index \"" + std::to_string(column_index) +
+                                    "\" type undefined.");
+    }
+    if (type_id == +model::TypeId::kMixed) {
+        throw std::invalid_argument("Column with index \"" + std::to_string(column_index) +
+                                    "\" contains values of different types.");
+    }
+    return column.GetType().IsMetrizable();
+}
+
+double DDVerifier::CalculateDistance(model::ColumnIndex const column_index,
+                                     std::pair<std::size_t, std::size_t> const &tuple_pair) const {
+    double dif = 0;
+    model::TypedColumnData const &column = typed_relation_->GetColumnData(column_index);
+    std::byte const *first_value = column.GetValue(tuple_pair.first);
+    std::byte const *second_value = column.GetValue(tuple_pair.second);
+    auto const &type = static_cast<model::IMetrizableType const &>(column.GetType());
+    dif = type.Dist(first_value, second_value);
+    return dif;
```

Comment:
We can just return that:
```suggestion
    return type.Dist(first_value, second_value);
```